apps/ui/src/components/Header.tsx

```diff
@@ -17,49 +17,184 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import { Button, ConfigProvider, Flex, Image, Layout, Typography } from 'antd';
+import { Button, ButtonProps, ConfigProvider, Drawer, Flex, Image, Layout, Typography, theme } from 'antd';
 
 import PCGL from '@/assets/pcgl-logo-full.png';
 import { pcglHeaderTheme } from '@/components/providers/ThemeProvider';
+import { Breakpoints, useMinWidth } from '@/global/hooks/useMinWidth';
+import { CloseOutlined, MenuOutlined } from '@ant-design/icons';
+import React, { useState } from 'react';
 
 const { Link } = Typography;
 const { Header } = Layout;
+const { useToken } = theme;
+
+interface MenuLinkType {
+	name: string;
+	isButton?: boolean;
+	buttonProps?: ButtonProps;
+	onClickAction?: VoidFunction;
+	href?: string;
+	position: 'left' | 'right';
+}
 
 const linkStyle: React.CSSProperties = {
 	minWidth: 100,
 	textAlign: 'center',
 };
 
 const HeaderComponent = () => {
+	const minWidth = useMinWidth();
+	const isResponsiveMode = minWidth <= Breakpoints.XL;
+
+	const { token } = useToken();
+
+	const [responsiveMenuOpen, setResponsiveMenuOpen] = useState(false);
+
+	const menuButtonStyle: React.CSSProperties = {
+		width: isResponsiveMode ? '100%' : 'auto',
+	};
+
+	/**
+	 * Default action when a button in the menu is clicked, used particularly for the mobile menu which should close after click.
+	 * @param buttonAction The function for the action needed to be preformed.
+	 */
+	const onMenuButtonClick = (buttonAction?: VoidFunction) => {
+		if (!buttonAction) {
+			return;
+		}
+		if (isResponsiveMode) {
+			setResponsiveMenuOpen(false);
+			buttonAction();
+		} else {
+			buttonAction();
+		}
+	};
+
+	const onLoginClick = () => {
+		console.info('Login Clicked. Not implemented yet.');
+	};
+
+	const menuLinks: MenuLinkType[] = [
+		{
+			name: 'Policies & Guidelines',
+			href: '#',
+			position: 'left',
+		},
+		{
+			name: 'Help Guides',
+			href: '#',
+			position: 'left',
+		},
+		{
+			name: 'Controlled Data Users',
+			href: '#',
+			position: 'left',
+		},
+		{
+			name: 'Apply For Access',
+			href: '#',
+			position: 'right',
+		},
+		{
+			name: 'Login',
+			isButton: true,
+			onClickAction: onLoginClick,
+			buttonProps: { color: 'primary', variant: 'solid' },
+			position: 'right',
+		},
+	];
+
+	/**
+	 * Generates the links to display in the mobile and desktop menus.
+	 * @param menuLinks An array containing the list of links for the menu
+	 * @param position The 'side' of links you want in the menu, left (next to the logo) or right (away from the logo on the other side of the screen)
+	 * @returns JSX Element array containing the link elements.
+	 */
+	const displayMenuLinks = (menuLinks: MenuLinkType[], position: 'left' | 'right' | 'both'): JSX.Element[] => {
+		return menuLinks
+			.filter((ml) => (position !== 'both' ? ml.position === position : ml.position))
+			.map((ml, key) =>
```

Comment:
Generally would prefer `link` or `menuLink` over `ml`, short character names can become 'unsearchable' when a project becomes sufficiently large. Descriptive names provide useful context.

Reply:
Also -- variable is `ml`, key is `mi-${key}`, inconsistent and I'm not sure what `mi` signifies

Reply:
good catch! that was def a typo, fixed!